ADR/ADR044-translate-api-response-at-request-time.md

```diff
@@ -0,0 +1,336 @@
+  "created_at": "2025-03-27T12:05:49.395300Z",
+  "updated_at": "2025-07-29T14:35:44.423406Z",
+  "creator_id": 1,
+  "what_happens_next_markdown": null,
+  "payment_url": null,
+  "submission_type": "email",
+  "share_preview_completed": false,
+  "s3_bucket_name": null,
+  "s3_bucket_aws_account_id": null,
+  "s3_bucket_region": null,
+  "language": "en",
+  "start_page": 33,
+  "steps": [
+    {
+      "id": 33,
+      "position": 1,
+      "next_step_id": null,
+      "type": "question_page",
+      "data": {
+        "question_text": "Llwythwch eich ffeil i fyny",
+        "hint_text": "",
+        "answer_type": "file",
+        "is_optional": false,
+        "answer_settings": null,
+        "page_heading": "Adroddiad blynyddol",
+        "guidance_markdown": "Peth arweiniad",
+        "is_repeatable": false
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+Advantages:
+- Request size doesn't increase
+- API structure doesn't change aside from the addition of a query parameter.
+- Keeps the translation logic out of runner
+Disadvantages:
+- Requires more logic in the API to choose the right translation - we already have some conversion logic, but we're currently trying to reduce the amount of logic in this repo.
+- API is doing more work at request time, which might slow down its responses for forms with lots of questions.
+- Small runner change required to request the correct version
+
+### 3. Creating separate Welsh and English snapshots when making a form live
+We could create separate snapshots for Welsh and English when making a form live - so the API would store two separate form documents, one in English and one in Welsh. At request time, the runner would request a form document and request a language, and API would return the relevant version of the form document. The objects returned would be identical to those in the 'Selecting languages at request time' approach.
+
+Advantages:
+- API structure doesn't change aside from the addition of a query parameter.
+- All translation logic including fallbacks can be handled within admin, API/runner don't have to do any processing on the resulting form documents - just need to serve/request the right version.
+- No additional processing or conversion would be required at request time, just returning an existing form document.
+Disadvantages:
+- More complexity in forms-admin. Probably needs API version increment.
+- Potential for overlap with API team's work.
+- Small runner change required to request the correct version.
+- Having two separate documents might risk the two versions going out of sync unexpectedly.
+- Would need to think about how we store the separate form documents - having them in one database column might be hard to query.
+
+## Decision
+
+Go with 'Selecting languages at request time' for now. Consider moving to 'Creating separate Welsh and English snapshots when making a form live' approach later if we add other languages and this this will be useful.
```

Comment:
Agree with "selecting languages at request time" - seems a sensible approach as requires minimal API schema changes and keeps response sizes the same. I wouldn't worry about making form-api do more work - that's a solvable problem, rather than a constraint. This also allows us to implement a non-breaking change to the API, i.e. just have it return English by default (e.g. when no locale parameter) specified.